### PR TITLE
feat(skyrim-platform): extend texts API 

### PR DESCRIPTION
**Pull request by VitaliyMubarakov**
# Extend texts API

image

### Setters

- ```skyrimPlatform.setTextPos(textId, xpos: float, ypos: float)``` - set the position of the text by id.

- ```skyrimPlatform.setTextString(textId, text: string)``` - set text by id.

- ```skyrimPlatform.setTextColor(textId, ["array of RBGA colors 0-1"])``` - set text color.

- ```skyrimPlatform.setTextSize(textId: number, size: float)``` - set text (not font) size.

- ```skyrimPlatform.setTextRotation(textId, rotation: float)``` - set text rotation.

- ```skyrimPlatform.setTextFont(textId, name: string)``` - set text font from relative path "Data/Platform/Fonts/", by name — "Tavern".

- ```skyrimPlatform.setTextDepth(textId, depth: int)``` - set text z-index of the text.

- ```skyrimPlatform.setTextEffect(textId, effect: skyrimPlatform.SpriteEffects)``` - set sprite effect [None = 0, FlipHorizontally = 1, FlipVertically = 2]

- ```skyrimPlatform.setTextOrigin(textId, origin [x,y])``` - set text pivot (center point).

### Getters

- ```skyrimPlatform.getTextPos(textId)``` - returns the coordinates(position) of the next as an array.

- ```skyrimPlatform.getTextString(textId)``` - returns a string.

- ```skyrimPlatform.getTextColor(textId)``` - returns an array of colors in RGBA.

- ```skyrimPlatform.getTextSize(textId)``` - returns the size of the text

- ```skyrimPlatform.getTextRotation(textId)``` - returns the rotation of the text

- ```skyrimPlatform.getTextFont(textId)``` - returns the path to the font

- ```skyrimPlatform.getTextDepth(textId)``` - returns z-index of the text

- ```skyrimPlatform.getTextEffect(textId)``` - returns effect enum.

- ```skyrimPlatform.getTextOrigin(textId)``` - returns pivot (center point) of the text

- ```skyrimPlatform.getNumCreatedTexts()``` - returns the number of created texts.

### CODE REVIEW

**Comment by Korisss** [comment]
super ficha srazy v prod

**Comment by Pospelove** [comment]
@Wedmer could you please take a look?
previous PR for reference: https://github.com/skyrim-multiplayer/skymp/pull/1114

File: skymp5-client/src/features/loadOrder.ts [hunk 1]
```diff
@@ -6,151 +6,151 @@ import { getScreenResolution } from "../view/formView";
...
+                );
+                return;
+            }
+            updateText(
+                'LOAD ORDER ERROR!\nCheck console for details.',
+                [255, 0, 0, 1]
+            );
+            sp.browser.loadUrl('about:blank');
+        });
 };
```
Comment (Wedmer) [inline skymp5-client/src/features/loadOrder.ts:156]: Do we need this indentation change? It's hard to check actual changes.

**Review by Wedmer — COMMENTED** [review]
Looks legit but needs testing.
Also check code formats, indentations and line endings.

**Review by Wedmer — CHANGES_REQUESTED** [review]
Please resolve format issues.

File: skymp5-client/src/features/loadOrder.ts (hunk 1 above)
Comment (VitaliyMubarakov) [reply skymp5-client/src/features/loadOrder.ts]: > Do we need this indentation change? It's hard to check actual changes.

fixed

File: skymp5-client/src/view/formView.ts [hunk 2]
```diff
@@ -17,438 +17,438 @@ import { lastTryHost, tryHost } from "./hostAttempts";
 import { ModelApplyUtils } from "./modelApplyUtils";
 
 export interface ScreenResolution {
-  width: number;
-  height: number;
+    width: number;
+    height: number;
```
Comment (Wedmer) [inline skymp5-client/src/view/formView.ts:21]: Indentation.

File: skymp5-client/src/view/formView.ts [hunk 3]
```diff
@@ -17,438 +17,438 @@ import { lastTryHost, tryHost } from "./hostAttempts";
...
+            const worldOrCell = ObjectReferenceEx.getWorldOrCell(Game.getPlayer() as Actor);
+            if (
+                worldOrCell !== 0 &&
+                model.movement.worldOrCell !== worldOrCell
+            ) {
+                this.destroy();
+                this.refrId = 0;
+                return;
+            }
+        }
```
Comment (Wedmer) [inline skymp5-client/src/view/formView.ts:68]: Indentation.

File: skymp5-client/src/view/formView.ts [hunk 4]
```diff
@@ -17,438 +17,438 @@ import { lastTryHost, tryHost } from "./hostAttempts";
...
+            !isBadMenuShown()
+        ) {
+            // Do not let actors breaking their equipment via inventory apply
+            // However, actually, actors do not have inventory in their models
+            // Except your clone.
+            if (!Actor.from(refr)) {
+                ModelApplyUtils.applyModelInventory(refr, model.inventory);
+                model.inventory = undefined;
+            }
+        }
```
Comment (Wedmer) [inline skymp5-client/src/view/formView.ts:216]: Indentation.

File: skymp5-client/src/view/formView.ts [hunk 5]
```diff
@@ -17,438 +17,438 @@ import { lastTryHost, tryHost } from "./hostAttempts";
...
 export const getScreenResolution = (): ScreenResolution => {
-  if (!_screenResolution) {
-    _screenResolution = {
-      width: Utility.getINIInt("iSize W:Display"),
-      height: Utility.getINIInt("iSize H:Display"),
+    if (!_screenResolution) {
+        _screenResolution = {
+            width: Utility.getINIInt("iSize W:Display"),
+            height: Utility.getINIInt("iSize H:Display"),
+        }
```
Comment (Wedmer) [inline skymp5-client/src/view/formView.ts:30]: Indentation

**Review by Wedmer — CHANGES_REQUESTED** [review]
Please follow original codestyle in typescript files. 
Base indentation is two space symbols.
Also something went wrong here:
```
[EOF Rule] File "docs/skyrim_platform/texts.md" has lint errors
[EOF Rule] File "skyrim-platform/src/tilted/ui/DX11RenderHandler.cpp" has lint errors
```

**Comment by VitaliyMubarakov** [comment]
> Please follow original codestyle in typescript files. Base indentation is two space symbols. Also something went wrong here:
> 
> ```
> [EOF Rule] File "docs/skyrim_platform/texts.md" has lint errors
> [EOF Rule] File "skyrim-platform/src/tilted/ui/DX11RenderHandler.cpp" has lint errors
> ```

- Ready, but we need formatting check (crlf/LF)

**Comment by VitaliyMubarakov** [comment]
Added a new line at end of the files. I hope the formatting error has been fixed

**Review by Wedmer — APPROVED** [review]
Looks much better. @Pospelove your turn.

**Review by Pospelove — APPROVED** [review]

**Comment by Pospelove** [comment]
Thanks for the PR <3

**Pull request merged**